CIScripts/Common/Invoke-NativeCommand.Tests.ps1

```diff
@@ -0,0 +1,121 @@
+Param (
+    [Parameter(Mandatory=$true)] [string] $TestenvConfFile
+)
+
+. $PSScriptRoot/Init.ps1
+. $PSScriptRoot/../Testenv/Testenv.ps1
+. $PSScriptRoot/VMUtils.ps1
+
+. $PSScriptRoot/Invoke-NativeCommand.ps1
+
+Describe "Invoke-NativeCommand" {
+    Context "Local machine" {
+        It "does not throw on successful command" {
+            Invoke-NativeCommand { whoami.exe }
+        }
+
+        It "can accept the scriptblock also as an explicit parameter" {
+            Invoke-NativeCommand -CaptureOutput -ScriptBlock { whoami.exe }
+        }
+
+        It "throws on failed command" {
+            { Invoke-NativeCommand { whoami.exe /invalid_parameter } } | Should Throw
+        }
+
+        It "throws on nonexisting command" {
+            { Invoke-NativeCommand { asdfkljasdsdf.exe } } | Should Throw
+        }
+
+        It "captures the exitcode of successful command" {
+            $Result = Invoke-NativeCommand -AllowNonZero { whoami.exe }
+            $Result.ExitCode | Should Be 0
+        }
+
+        It "captures the exitcode a failed command" {
+            (Invoke-NativeCommand -AllowNonZero { whoami.exe /invalid }).ExitCode | Should Not Be 0
+        }
+
+        It "can capture the output of a command" {
+            (Invoke-NativeCommand -CaptureOutput { whoami.exe }).Output | Should BeLike '*\*'
+        }
+
+        It "can capture multiline output" {
+            (Invoke-NativeCommand -CaptureOutput { whoami.exe /? }).Output.Count | Should BeGreaterThan 1
+        }
+
+        It "does not capture the output by default" {
+            Invoke-NativeCommand { whoami.exe } | Should BeNullOrEmpty
+        }
+
+        It "can use variables it scriptblock" {
+            $Command = "whoami.exe"
+            (Invoke-NativeCommand -CaptureOutput -AllowNonZero { & $Command }).ExitCode | Should Be 0
+        }
+
+        It "preserves the ErrorActionPreference" {
+            $ErrorActionPreference = "Stop"
+            Invoke-NativeCommand -CaptureOutput { whoami.exe } | Out-Null
+            $ErrorActionPreference | Should Be "Stop"
```

Comment:
I know that `$ErrorActionPreference` is scoped in this case, but in the function itself (line 79) we set it to stop. Can we maybe use a different `$ErrorActionPreference` in this test just so we can make sure that we actually prove that it is indeed handled correctly?

Reply:
Perhaps use two testcases with `Stop` and `Continue`?